### DISCUSSION TITLE
Question - GIF Creations Instead Of Only Images

### DISCUSSION THREAD

**Discussion by Roebou1989** [nftchef/art-engine#115] [Help + Q&A]
Hi @nftchef ,

I have been searching far and wide, but not getting any direction on this topic. I have a user that would like to be able to use a layer setup where they have both gifs and png files as their layer items, all with transparent backgrounds, apart from the backgrounds folder, and would like to build a new gif that is made up of the combination of these layers.

Have you been able to build this on your side before or have seen anyone one build it or can perhaps assist me in building this out? I am attaching a layer configuration set if you would like to take a stab at it.

[Layer.zip](https://github.com/nftchef/art-engine/files/8255495/Layer.zip)

Thanks for all your work on the code repo, much appreciated!

Kind Regards,
Rohan (ThePeanutGalleryAndCo) ;)

**Comment by nftchef** [comment]
In order to do this you will need some customization to the engine or the input artwork. There is a good example of gif's from sprite sheets in this repo: https://github.com/MichaPipo/Generative_Gif_Engine

To do it with gif's and png's, you will need to add some custom handling features that can auto-create sprite sheets while doing the image generation and compile it all back to a gif in the end.

**Comment by Roebou1989** [comment]
Thanks for the feedback, appreciated! Do you think you will attempt this on your repo? If yes, then I will leave it to the experts and apply that into my code base, otherwise, I will give it another go, but some of the sprites and info went a tad bit over my head from what I read. Most of the people simply create sprites from static images, I haven't seen ones where they decompose GIF files and re-compose ot again when building up a new GIF.

One of the more tougher head scratchers that I have come across 😀

**Reply by nftchef** [reply]
While it's something I've wanted to do for a while, I don't have any immediate plans to incorporate it into this codebase (unless a project comes along that requires it). It's also possible to start with sprite sheets and go the route you described, which would more or less use the generator _as is_ and require steps later to compose the sprites back into gifs

**Comment by jalagar** [comment]
Hi @thepeanutgalleryandco and @nftchef I am working on expanding on MichaPipo's repo to work for separate png -> spritesheet -> rarity -> png -> gif. I found issues with MichaPipo's javascript method of generating gifs by using the js libraries and navigating pixel by pixel where sometimes the colors are wrong. It is also very slow (4 minutes to generate 15 gifs at the highest quality) and take significant memory and CPU. I switched to using [python PIL ](https://github.com/python-pillow/Pillow) and I can generate 100 gifs in under 30s without losing quality. 

I plan to write a medium post and open source the repo later this week and will share it here! Hopefully as a v2 I can incorporate nftchef's logic (or vice versa) to work for if-then statements and layers! We need it for my collection that I'm launching 👀. Feel free to follow me on twitter https://twitter.com/jalagar_eth for updates as well.

**Comment by jalagar** [comment]
Hi @thepeanutgalleryandco @nftchef As promised:

Here's a medium article: https://jalagar-eth.medium.com/how-to-create-generative-animated-nft-art-in-under-an-hour-e7dab1785c56
And here's the repo: https://github.com/jalagar/Generative_Gif_Engine
Feel free to follow me on twitter: https://twitter.com/jalagar_eth for more updates.

You can see all 1k of them on https://opensea.io/collection/genesis-bouncing-ball. Let me know if you have any questions!

**Comment by Roebou1989** [comment]
@jalagar this issue has never been taking png files and generating a gif out of them. The art engine could already do that. The issue has always been taking png files and gif files, then generating new gifs from these files. 

Can your code base do this?

**Comment by jalagar** [comment]
Hi @thepeanutgalleryandco yes so you will need to split your gif into the individual layers (using something like https://ezgif.com/split). then it will combine the png layers and spit out the generative gifs for you. Let me know if the README in the repo doesn't make sense or work for your use case I might have misunderstood.

**Comment by jalagar** [comment]
As an enhancement, I might be able to take in the gif directly and split it into the individual layers, I added an issue to track this https://github.com/jalagar/Generative_Gif_Engine/issues/4

**Comment by jalagar** [comment]
@thepeanutgalleryandco I was able to integrate nftchef's features into my repo https://github.com/jalagar/Generative_Gif_Engine
![0](https://user-images.githubusercontent.com/98709501/162953217-f800b75d-0c5e-4e38-badb-f29842b2cb96.gif)
